src/allmydata/test/cli/test_invite.py

```diff
@@ -0,0 +1,187 @@
+import os
+import mock
+import json
+from os.path import join
+
+from twisted.trial import unittest
+from twisted.internet import defer
+from ..common_util import run_cli
+from ..no_network import GridTestMixin
+from .common import CLITestMixin
+
+
+class _FakeWormhole(object):
+
+    def __init__(self, outgoing_messages):
+        self.messages = []
+        self._outgoing = outgoing_messages
+
+    def get_code(self):
+        return defer.succeed(u"6-alarmist-tuba")
+
+    def set_code(self, code):
+        self._code = code
+
+    def get_welcome(self):
+        return defer.succeed(
+            json.dumps({
+                u"welcome": {},
+            })
+        )
+
+    def allocate_code(self):
+        return None
+
+    def send_message(self, msg):
+        self.messages.append(msg)
+
+    def get_message(self):
+        return defer.succeed(self._outgoing.pop(0))
+
+    def close(self):
+        return defer.succeed(None)
+
+
+def _create_fake_wormhole(outgoing_messages):
+    return _FakeWormhole(outgoing_messages)
+
+
+class Join(GridTestMixin, CLITestMixin, unittest.TestCase):
+
+    @defer.inlineCallbacks
+    def setUp(self):
+        self.basedir = self.mktemp()
+        yield super(Join, self).setUp()
+        yield self.set_up_grid(oneshare=True)
+
+    @defer.inlineCallbacks
+    def test_create_node_join(self):
+        node_dir = self.mktemp()
+
+        with mock.patch('allmydata.scripts.create_node.wormhole') as w:
+            fake_wh = _create_fake_wormhole([
+                json.dumps({u"abilities": {u"server-v1": {}}}),
+                json.dumps({
+                    u"needed": 1,
+                    u"happy": 1,
+                    u"total": 1,
+                    u"nickname": u"somethinghopefullyunique",
+                    u"introducer": u"pb://foo",
+                }),
+            ])
+            w.create = mock.Mock(return_value=fake_wh)
+
+            rc, out, err = yield run_cli(
+                "create-client",
+                "--join", "1-abysmal-ant",
+                node_dir,
+            )
```

Comment:
Hmmm.  I want to suggest making `wormhole` a parameter to all of the APIs that otherwise need it monkey-patched in from the tests.... but `run_cli` seems to get in the way of that idea.  That's too bad. :/